### PR TITLE
[INLONG-10721][Sort] Add elasticsearch7 connector on flink 1.18

### LINKED ISSUE #10721: [Feature][Sort] Add Elasticsearch7 connector on Flink 1.18
### Description

Add Elasticsearch7 connector on Flink 1.18 to sort-flink-v1.18
The parent issue is #10467 

### Use case

User can sink data to elasticsearc7 through this connector.

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by XiaoYou201**
### [INLONG-10721][Sort] Add elasticsearch7 connector on flink 1.18.

Fixes #10721

### Motivation

There is no es connector in sort-connectors， but es is a popular distributed, RESTful search and analytics engine.

So, this pr will add the es7 connector onflink 1.18.

### Modifications

Reference the official connector flink-connector-elasticsearch7 and it will support send audit information later.

### Verifying this change

0b421912daeef5e98769337f17a4b0e3_compress
d879186a9a61fa69e02999cd4bc5fe5b_compress
569759ac3a6b5838dad5cbddd43297d2_compress

### Documentation

  - Does this pull request introduce a new feature? (yes)

### CODE REVIEW

**Review by EMsnap — COMMENTED** [review]
please add licenses

**Comment by XiaoYou201** [comment]
> please add licenses

done～

**Review by EMsnap — APPROVED** [review]

File: inlong-sort/sort-flink/sort-flink-v1.18/sort-connectors/elasticsearch7/pom.xml [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+        <elasticsearch.connector.version>3.0.1-1.17</elasticsearch.connector.version>
+    </properties>
+
+    <dependencies>
+        <dependency>
+            <groupId>org.apache.inlong</groupId>
+            <artifactId>sort-flink-dependencies-v1.18</artifactId>
+            <version>${project.version}</version>
+            <scope>provided</scope>
+        </dependency>
```
Comment (aloyszhang) [inline inlong-sort/sort-flink/sort-flink-v1.18/sort-connectors/elasticsearch7/pom.xml:43]: `sort-flink-dependencies-v1.18` is already dependent in the parent pom.

Comment (XiaoYou201) [reply inlong-sort/sort-flink/sort-flink-v1.18/sort-connectors/elasticsearch7/pom.xml:43]: The sort-flink-dependencies-v1.18 dependency scope is compile.But this dependencies is unnecessary for connector.When connector running in the cluster, the cluster already has this denpendencies.If add them, there will be error submitting job.

**Review by aloyszhang — APPROVED** [review]

**Review by aloyszhang — APPROVED** [review]

**Pull request merged**